src/ray/rpc/gcs_server/gcs_rpc_client.h

```diff
@@ -241,7 +241,7 @@ class GcsRpcClient {
           callback(status, reply);
         }
         delete executor;
-      } else if (!status.IsRpcError()) {
+      } else if (!IsGrpcRetryableStatus(status)) {
```

Comment:
GCS can return non-OK and non-retryable errors. Excerpt from the PR initial comment:

- GcsClient now only retry grpc UNAVAILABLE and not other codes, since we only want to retry on GCS down and not other cases (e.g. RESOURCE_EXHAUSTED). See python/ray/tune/tests/test_tune_restore.py::ResourceExhaustedTest::test_resource_exhausted_info

In this example, Ray Train made an extra large message to GCS, which exceeds gRPC limit. It can't possibly be sent and we should not retry.